sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { toBase64, toHex } from '@mysten/bcs';
+import { secp256r1 } from '@noble/curves/p256';
+import { blake2b } from '@noble/hashes/blake2b';
+import { randomBytes } from '@noble/hashes/utils';
+import type {
+	AuthenticationCredential,
+	RegistrationCredential,
+} from '@simplewebauthn/typescript-types';
+
+import { PasskeyAuthenticator } from '../../bcs/bcs.js';
+import type { IntentScope, SignatureWithBytes } from '../../cryptography/index.js';
+import { messageWithIntent, SIGNATURE_SCHEME_TO_FLAG, Signer } from '../../cryptography/index.js';
+import type { PublicKey } from '../../cryptography/publickey.js';
+import type { SignatureScheme } from '../../cryptography/signature-scheme.js';
+import {
+	parseDerSPKI,
+	PASSKEY_PUBLIC_KEY_SIZE,
+	PASSKEY_SIGNATURE_SIZE,
+	PasskeyPublicKey,
+} from './publickey.js';
+
+/**
+ * A passkey signer used for signing transactions. This is a client side implementation for [SIP-9](https://github.com/sui-foundation/sips/blob/main/sips/sip-9.md).
+ */
+export class PasskeyKeypair extends Signer {
+	private publicKey: Uint8Array;
+
+	/**
+	 * Get the key scheme of passkey,
+	 */
+	getKeyScheme(): SignatureScheme {
+		return 'Passkey';
+	}
+
+	/**
+	 * Creates an instance of Passkey signer. It's expected to call the static `getPasskeyInstance` method to create an instance.
+	 * For example:
+	 * ```
+	 * const signer = await AwsKmsSigner.getPasskeyInstance();
+	 * ```
+	 */
+	constructor(publicKey: Uint8Array) {
+		super();
+		this.publicKey = publicKey;
+	}
+
+	/**
+	 * Creates an instance of Passkey signer invoking the passkey from navigator.
+	 */
+	static async getPasskeyInstance(): Promise<PasskeyKeypair> {
+		// create a passkey secp256r1 key for the relying party.
+		const credential = (await navigator.credentials.create({
```

Comment:
@Jordan-Mysten pointed out this won't work in our mobile wallet, because the navigator api doesn't exist in react-native.

What we should do instead is create a minimal interface with the required methods (`create`, and `get`) that can be passed in anywhere you can create a passkey keypair instance (we do something similar for zklogin public identifiers which take a GraphQL client instance for signature verification). 

We would then have a basic default implementation that uses the navigator api, but we would have an easy path for using the native passkey apis in the mobile wallet

Reply:
good catch. lemme abstract this into an interface. not familiar with react native - is this something we should figure out now to ensure compatibility or we can add later?

Reply:
I think it should be fine to add later, as long as the interface is generic enough that it should work with other passkey apis.

first result for a react-native implementation is this: https://github.com/f-23/react-native-passkey, it looks like it has get/create methods that should work the same way